python/cudf/cudf/core/series.py

```diff
@@ -733,9 +733,9 @@ def from_pandas(cls, s: pd.Series, nan_as_null=no_default):
             result = cls(s, nan_as_null=nan_as_null)
         return result
 
-    @property  # type: ignore
+    @functools.cached_property
```

Comment:
*issue*: `Series` objects can be updated "in-place" so you'd either want to invalidate these cached properties, or else flip back to `@property`